gradle-plugin/src/main/java/com/microsoft/gradle/GradleProjectModelBuilder.java

```diff
@@ -142,6 +180,25 @@ private DefaultGradleDependencyNode resolveDependency(ResolvedDependencyResult r
 		return closures;
 	}
 
+	private List<GradleTask> getGradleTasks(Project rootProject, Project project) {
```

Comment:
This is not entirely correct as some of the tasks might not be present in the container at this point. You'd need to call `InternalTaskContainer.discoverTasks()` and `InternalTaskContainer.realize()`: https://github.com/gradle/gradle/blob/ab798e698eb94b6c0ae67011016c67f77b74b601/subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/GradleProjectBuilder.java#L100-L101

It might be a better strategy to reuse the built-in `GradleProject` Tooling API model. You can define a custom TAPI build action that runs two model queries(GradleProject and GradleProjectModel) in one request.

Reply:
Thanks a lot for the information! I'll track it in #1168 and fix it in our next release.